### PR TITLE
connectivity: Allow egress to DNS by CIDR

### PR DESCRIPTION
**Pull request by sathieu**
This is needed for node-local-dns installed by kubespray, using `hostNetwork`.

See
https://github.com/kubernetes-sigs/kubespray/blob/039205560a5a38dac7e180ec4c46ce4edd404d39/roles/kubernetes-apps/ansible/templates/nodelocaldns-daemonset.yml.j2

See also #995 and https://github.com/cilium/cilium-cli/pull/997.

### CODE REVIEW

**Review by tklauser — APPROVED** [review]
Thanks

**Comment by sathieu** [comment]
Marking the PR as draft, as matching by label doesn't work.

The nodelocaldns pod is seen as `world`.

**Comment by sathieu** [comment]
> Marking the PR as draft, as matching by label doesn't work.
> 
> The nodelocaldns pod is seen as `world`.

I've taken another path for this (by IP). Need to test before removing draft status

**Comment by sathieu** [comment]
@eminaktas  superseeded by e6ef23e362905e22730b2029f6cc682e36ae609c? Or should I rebase and resolve conflicts?

**Comment by eminaktas** [comment]
My PR resolves the same issue as well. However, you can rebase and resolve the conflicts and replace my changes with your solution.

As a next step, we need to investigate Cilium itself to resolve the actual problem. https://github.com/cilium/cilium/issues/18644#issuecomment-1062265283

**Comment by sathieu** [comment]
Rebased/improved on top of https://github.com/cilium/cilium-cli/pull/997.

@tklauser Please review again :pray:!

File: connectivity/manifests/client-egress-only-dns.yaml [hunk 1]
```diff
@@ -15,16 +15,23 @@ spec:
...
-  # test for such environments.
+    - matchLabels:
+        io.kubernetes.pod.namespace: kube-system
+        k8s-app: kube-dns
+    - matchLabels:
+        io.kubernetes.pod.namespace: kube-system
+        k8s-app: coredns
+    - matchLabels:
+        io.kubernetes.pod.namespace: kube-system
+        k8s-app: node-local-dns
```
Comment (tklauser) [inline connectivity/manifests/client-egress-only-dns.yaml:26]: Why is this changed only in this policy but not the others? It also looks like `k8s-app: nodelocaldns` is missing.

Comment (eminaktas) [reply connectivity/manifests/client-egress-only-dns.yaml:26]: You should leave this part as it was.

Comment (sathieu) [reply connectivity/manifests/client-egress-only-dns.yaml]: Done.

@tklauser @eminaktas please review again.

File: connectivity/manifests/allow-all-except-world-pre-v1.11.yaml [hunk 2]
```diff
@@ -13,16 +13,14 @@ spec:
     - health
   - toEndpoints:
     - {}
-  # When node-local-dns is deployed with local IP,
-  # Cilium labels its ip as world.
-  # This change prevents failing the connectivity
-  # test for such environments.
+  {{- if .DNSCIDR }}
```
Comment (tklauser) [inline connectivity/manifests/allow-all-except-world-pre-v1.11.yaml:16]: I'm not sure we should make this dependent on .DNSCIDR being set. Shouldn't we keep the existing policy in case the user didn't specify .DNSCIDR/--dns-cidr?

**Review by tklauser — COMMENTED** [review]
It looks like the PR title and description no longer match the change. Could you please update these?

I've also requested a review from @cilium/sig-k8s as I'm not terribly familiar with node-local DNS myself.

File: connectivity/manifests/allow-all-except-world-pre-v1.11.yaml (hunk 2 above)
Comment (sathieu) [reply connectivity/manifests/allow-all-except-world-pre-v1.11.yaml:16]: I don't know, but https://github.com/cilium/cilium-cli/pull/997 is not released (yet), so this is not a regression.

**Comment by sayboras** [comment]
Removed myself from reivewer as it could get added unintentionally.

**Comment by tklauser** [comment]
> Removed myself from reivewer as it could get added unintentionally.

@sayboras nope, that was intentional ;-) I've [requested](https://github.com/cilium/cilium-cli/pull/1341#event-8423111520) a review from @cilium/sig-k8s (also see https://github.com/cilium/cilium-cli/pull/1341#pullrequestreview-1281409837) and it seems you were assigned. Do you know who would be a good reviewer with knowledge about node-local-dns and respective policies?

**Comment by sayboras** [comment]
> Do you know who would be a good reviewer with knowledge about node-local-dns and respective policies?

yeah, node-local-dns is kind of de factor for every EKS clusters, which I have some prior experience. Let me take a look :eyes:.

**Review by sayboras — APPROVED** [review]
Thanks :100:

**Comment by sathieu** [comment]
Abandoning this PR.
